thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1381,7 +1386,7 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
             std::atomic<IKeyManager *> & keyManager = (*keyManagers)[selected];
             if (!keyManager) // delayed until actually needed
             {
-                keyManager = activity.createPartKeyManager(partNo, copy);
+                keyManager = activity.createPartKeyManager(partNo, copy, &contextLogger);
```

Comment:
There's another problem here.
A lookup handler may be dealing with >1 part, but there is only one CStatsContextLogger involved here.
Which means in getFileStats below, it will iterate through the IKeyManager's (1 per part), and merge stats from the shared context logger used by all, causing stats to be over reported.